File: app/schemas/case_progress.py
from pydantic import BaseModel


class CaseProgressOut(BaseModel):
    case_id: int
    total: int
    approved: int
    uploaded: int
    rejected: int
    required: int
    percent: int  # 0..100
    is_ready_to_submit: bool
    is_ready_for_approval: bool

    class Config:
        from_attributes = True

File: veteran-aid-web/src/api/case-progress.ts
import { http } from "@/api/http";

export type CaseProgress = {
  case_id: number;
  total: number;
  approved: number;
  uploaded: number;
  rejected: number;
  required: number;
  percent: number;
  is_ready_to_submit: boolean;
  is_ready_for_approval: boolean;
};

export async function fetchCaseProgress(caseId: number): Promise<CaseProgress> {
  const res = await http.get<CaseProgress>(`/cases/${caseId}/progress`);
  return res.data;
}
